### DISCUSSION TITLE
Buddy software improvements

### DISCUSSION THREAD

**Discussion by g7uhn** [g7uhn/ft817_buddy#30] [Ideas]
Are you any good with Arduino code?  There are some obvious improvements that could be made to the Buddy's software and it would be great to get some help with the code.

The main limitation at the moment (April 2021) is the very basic polling routine that polls the radio and checks for button inputs in between polls... it's good enough to work but operation could be a lot smoother.  Is threading the answer?  Can you help re-work it?

With improvements to the structure of the code it should be possible to sneak in a few more features like Power/SWR meter monitoring and speed up the response to button pressing.

If you can help, get in touch!  😄

**Comment by ta33ers** [comment]
I hope this makes its way to you. My name is Glendon, call sign KG5WXU in Fort Worth, Texas. I was going through a box of projects when I found the PC boards I had purchased for you 817 Buddy. I decided it was about time to assemble them, so I ordered the parts I needed, and got to the K16 key chip, and found that they are now only shipping chips that are 3.3 volt. Will they work on the 817 Buddy, or will a modification need to be made? As for the Arduino code, I'm not the code master I'd like to think I am, but I sometimes surprise myself, so that will be my next project. When I started all this I didn't bother to read that this was 817 specific (I have an 818), but I guess if I'm forced to buy more ham gear I'll just have to deal with it🙀😹 I hope to hear from you, thank you for your time Glendon KG5WXU 73

**Comment by g7uhn** [comment]
Hey Glendon, thanks for reaching out!  So cool to hear from people who like
this little project :-D

Wow, 3.3V on the latest K16 keyer chips?  Hmmm, I didn't see that coming.
Modifying the PCBs you have to work with that might be possible but
tricky... perhaps using the pads for C6 to build a potential divider with
two resistors (to create the 3.3V from the 5V Vdd) and then wiring that
3.3V back to Pin 1 of the K16 chip holder (making sure Pin 1 of the chip
holder isn't connected to the 5V Vdd rail, probably cutting or bending that
leg of the chip holder).  Does that sound like something you'd be
comfortable with?

As far as FT-818 compatibility goes, I've heard of a few people lining up
to try this project on an 818 but no one has told me it does or doesn't
work yet!  I've just pinged one of them to see if he's got round to
finishing his project and trying it, I'll let you know what he says.

Thanks again for dropping me a line, great to hear from you!

Andy
G7UHN
